content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -92,6 +92,13 @@ Name | Encryption | Strength | Speed | Key Length | Other Considerations
 Each provider supports multiple keys - the keys are tried in order for decryption, and if the provider
 is the first provider, the first key is used for encryption.
 
+__It is not recommended to use EncryptionConfig without using the `kms` provider, as this does not materially change your
```

Comment:
I think we should be a _little_ bit softer on the messaging here. Using an `EncryptionConfiguration` does increase security posture a bit, albeit not as well as an external kms provider.

**Storing the raw encryption key in the EncryptionConfig only moderately improves your security posture. Please use a `kms` provider for secure production workloads.** Encrypting secrets with a locally managed key protects against an etcd compromise, but it fails to protect against a host compromise. Since the encryption keys are stored on the host in the `EncryptionConfig` YAML file, a skilled attacker can access that file and extract the encryption keys. Kubernetes 1.10+ allows you to leverage an external KMS provider such that the key is stored in a completely separate system. An attacker would need to compromise etcd, the kubeapi-server, and the third-party KMS provider to retrieve the plaintext values, providing a higher level of security than locally-stored encryption keys.

Reply:
Thanks - took a bunch of your text, helpful to better explain threat model.

Reply:
@sethvargo curious, what will be `kms` for non public cloud envs ?